tower-http/src/compression/service.rs

```diff
@@ -12,19 +13,24 @@ use tower_service::Service;
 ///
 /// See the [module docs](crate::compression) for more details.
 #[derive(Clone, Copy)]
-pub struct Compression<S> {
+pub struct Compression<S, P = DefaultCompressionFilter> {
     pub(crate) inner: S,
     pub(crate) accept: AcceptEncoding,
+    pub(crate) compression_filter: P,
 }
 
 impl<S> Compression<S> {
     /// Creates a new `Compression` wrapping the `service`.
-    pub fn new(service: S) -> Self {
+    pub fn new(service: S) -> Compression<S> {
         Self {
             inner: service,
             accept: AcceptEncoding::default(),
+            compression_filter: DefaultCompressionFilter{}
         }
     }
+}
+
+impl<S, P: CompressionFilter> Compression<S, P> {
```

Comment:
Unless the trait bound is specifically necessary for lots of methods in the impl block then you commonly leave it. I would do

```suggestion
impl<S, P> Compression<S, P> {
```